Runtime/AvatarCreator/UI/SelfieElement.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Generic;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public class SelfieElement : MonoBehaviour
+    {
+        private const string TAG = nameof(SelfieElement);
+
+        [SerializeField] private ImageConfirmationElement imageConfirmationElement;
+        [SerializeField] private BodyType bodyType;
+        [SerializeField] private OutfitGender gender;
+
+        [Space(5)]
+        [Header("Events")]
+        public UnityEvent<GameObject,AvatarProperties> onAvatarCreated;
+
+        private void OnEnable()
+        {
+            imageConfirmationElement.onImageConfirmed.AddListener(OnPhotoCaptured);
+        }
+
+        private void OnDisable()
+        {
+            imageConfirmationElement.onImageConfirmed.RemoveListener(OnPhotoCaptured);
+        }
+
+        private async void OnPhotoCaptured(Texture2D texture)
+        {
+            var bytes = texture.EncodeToPNG();
+            var byteAsString = Convert.ToBase64String(bytes);
+
+            var avatarProperties = new AvatarProperties();
+            avatarProperties.Partner = CoreSettingsHandler.CoreSettings.Subdomain;
+            avatarProperties.Base64Image = byteAsString;
+            avatarProperties.BodyType = bodyType;
+            avatarProperties.Gender = gender;
+            avatarProperties.Assets = new Dictionary<Category, object>();
+
+            if (!AuthManager.IsSignedIn && !AuthManager.IsSignedInAnonymously)
+            {
+                SDKLogger.Log(TAG, "Not signed in");
+                return;
+            }
+            
+            var avatarManager = new AvatarManager(bodyType);
+            var avatar = await avatarManager.CreateAvatar(avatarProperties);
+
+            onAvatarCreated?.Invoke(avatar.Item1, avatar.Item2);
```

Comment:
What is avatar item1 and item2? Looking at this code I would have no idea what this means without stepping into it more.

Reply:
Though consumer of this event will know what those 2 elements are  
```
public UnityEvent<GameObject,AvatarProperties> onAvatarCreated;
```

Reply:
True, but still, I have no idea what gameObject is being passed, other than that it is called Item1. I think the code should be as self documenting as possible.

Although it looks like c# added a way to get better naming with tuples, so maybe there is a middle ground? https://learn.microsoft.com/en-us/archive/msdn-magazine/2015/december/essential-net-designing-csharp-7

Reply:
oh nice didn't knew that, will change it accordingly.